core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -812,4 +815,32 @@ private boolean deleteProperties(Namespace namespace, Set<String> properties) {
   protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
+
+  @Override
+  public TableBuilder buildTable(TableIdentifier identifier, Schema schema) {
+    return new ViewAwareTableBuilder(identifier, schema);
+  }
+
+  /**
+   * The purpose of this class is to add view detection only when SchemaVersion.V1 schema is used
+   * when replacing a table.
+   */
+  protected class ViewAwareTableBuilder extends BaseMetastoreCatalogTableBuilder {
```

Comment:
for other reviewers: the `BaseMetastoreViewCatalogTableBuilder` is already view-aware during a table replace, but the existing `TestJdbcCatalog` should be running with schema V0 (not view-aware) as default. This makes it necessary to override the default behavior in order to only perform the view check when `schemaVersion == JdbcUtil.SchemaVersion.V1`